server/controllers/items/recent_public.coffee

```diff
@@ -15,17 +15,23 @@ module.exports = (req, res)->
   .catch error_.Handler(req, res)
 
 selectRecentItems = (items)->
+  # Group items in an object of owners ID,
+  # then select only x firsts items from each owner
+  # turn object into flattened array
+
   itemsLimit = 15
   maxItemsPerOwner = 3
 
-  Promise.resolve _.groupBy items, (item)-> item.owner
-  .then (itemsByOwner)->
-    _.map itemsByOwner, (items)->
-      items.slice 0, maxItemsPerOwner
-  .then (limitedItemsByOwner)->
-    _.flatten _.values limitedItemsByOwner
-  .then (allRecentItems)->
-    allRecentItems.slice 0, itemsLimit
+  _(items)
+  .groupBy itemOwner
+  .map (items, _)-> items.slice(0, maxItemsPerOwner)
+  .flatten()
+  .values()
+  .slice 0, itemsLimit
+  .shuffle()
+  .value() # wrapping lodash .chain() function
+
```

Comment:
In this kind of chain, I generally prefer to have the comments directly on top of what then comment:
```coffee
  _(items)
  # Group items in an object of owners ID,
  .groupBy itemOwner
  # then select only x firsts items from each owner
  .map (items, _)-> items.slice(0, maxItemsPerOwner)
  # turn object into flattened array
  .flatten()
  .values()
  .slice 0, itemsLimit
  .shuffle()
  # wrapping lodash .chain() function
  .value()
```
examples: [1](https://github.com/inventaire/inventaire/blob/5b9bc9d/server/controllers/invitations/lib/parse_emails.coffee#L22-L28), [2](https://github.com/inventaire/inventaire/blob/5b9bc9d/server/controllers/groups/lib/slugify.coffee), [3](https://github.com/inventaire/inventaire/blob/5b9bc9d/server/controllers/entities/lib/scaffold_entity_from_seed/search_work_entity_by_title_and_authors.coffee#L32-L37)